### DISCUSSION TITLE
How to combine a thermal and grain growth model using multiapp

### DISCUSSION THREAD

**Discussion by ajyano22** [idaholab/moose#17708] [Q&A General] [answered]
Hi, I'm relatively new to MOOSE so forgive me if there is an obvious answer to this. I'm trying to combine the function_ellipsoid_heat_source model with a voronoi grain growth model. Both of these exist in the MOOSE library and I know they can be linked through use of multiapp, likely with the thermal model being the master and the grain growth as a sub-app. I'm just not sure how to get them to pull from their respective libraries. Do I need to be registering them in the makefile? I didn't want to have to create a separate app, but would that be the best way? 

Thank for the help in advance!

**Comment by permcody** [comment, marked as answer]
If you really don't want your own application. You can compile in the "modules" directory and get a "combined module" executable where you have access to all of the objects in all of the modules. You can then choose to setup a problem like this as a single monolith solve where everything is one nonlinear system (fully-coupled), or you could setup a Multiapp this way (both the master and sub would just be the combined module app) and do a loosely or tightly coupled solve between those two physics.

**Reply by ajyano22** [reply]
When you say a "combined module" executable do you mean like how if you compile in "combined" you get the combined-opt executable? Or is that the executable I should be using?

**Reply by ajyano22** [reply]
When running with the combined-opt it begins to run but then I get this error about an object not being registered which shouldn't they all be accessible in combined?


Running command: /home/aliyay/projects/moose/modules/combined/combined-opt Outputs/csv=true -i peacock_run_exe_tmp_function_heat_source_master.i
Working directory: /home/aliyay/projects/moose/modules/combined

Creating MultiApp sub_app of type MooseTestApp of level 1 and number 0:
sub_app0: 


*** ERROR ***
A 'PolycrystalVoronoi' is not a registered object.

If you are trying to find this object in a dynamically linked library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.

**Reply by GiudGiud** [reply]
If you don't want to build your own application, then yes, use the combined module to be able to use all moose objects currently existing

Indeed it should be registered, as PolyCrystalVoronoi is a phase field module object. Can you please attach the rest of the input file?
also does it run with the phase field module executable?

Guillaume
